joi-validations/users.validations.js

```diff
@@ -0,0 +1,12 @@
+const {object, string} = require('./general.validations');
+
+const UserValidation = {
+  authenticateWeb: {
+    body: object({
+      country_code: string.max(2).min(2).uppercase().required(),
+      human_id: string.required()
+    })
+  }
+};
+
+module.exports = {UserValidation};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The validation for `country_code` only checks if it's a string of exactly 2 characters and converts it to uppercase. However, it doesn't validate if it's a valid ISO country code. Consider adding a custom validation function that checks if the provided country code is in a list of valid ISO country codes.

```diff
-      country_code: string.max(2).min(2).uppercase().required(),
+      country_code: string.custom((value, helpers) => {
+        const validCountryCodes = ['US', 'GB', 'DE', ...]; // Add all valid country codes here
+        if (!validCountryCodes.includes(value)) {
+          return helpers.error('any.invalid');
+        }
+        return value;
+      }).max(2).min(2).uppercase().required(),
```

Also, consider adding length validation for `human_id`. Currently, any string regardless of its length will be accepted as long as it's not empty. This could potentially lead to performance issues if very large strings are passed.

```diff
-      human_id: string.required()
+      human_id: string.min(1).max(100).required() // Adjust min and max values according to your requirements
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->